tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/Desugarer.scala

```diff
@@ -51,16 +50,29 @@ class Desugarer(tracker: TransformationTracker) extends TlaExTransformation {
         val eqs = flatArgs map { x => tla.eql(tla.prime(x), x) }
         tla.and(eqs :_*)
 
-      case fex @ OperEx(TlaSetOper.filter, boundEx, setEx, predEx) =>
+      case OperEx(TlaSetOper.filter, boundEx, setEx, predEx) =>
         OperEx(TlaSetOper.filter, collapseTuplesInFilter(transform(boundEx), transform(setEx), transform(predEx)) :_*)
 
-      case fex @ OperEx(TlaSetOper.map, args @ _*) =>
+      case OperEx(TlaSetOper.map, args @ _*) =>
         val trArgs = args map transform
         OperEx(TlaSetOper.map, collapseTuplesInMap(trArgs.head, trArgs.tail) :_*)
 
-      case fex @ OperEx(TlaFunOper.funDef, args @ _*) =>
+      case OperEx(funDefOp, args @ _*) if (funDefOp == TlaFunOper.funDef || funDefOp == TlaFunOper.recFunDef) =>
         val trArgs = args map transform
-        OperEx(TlaFunOper.funDef, collapseTuplesInMap(trArgs.head, trArgs.tail) :_*)
+        val fun = trArgs.head
+        val (vars, sets) = TlaOper.deinterleave(trArgs.tail)
+        val (onlyVar, onlySet) =
+          if (vars.length > 1) {
+            val pair = (tla.tuple(vars :_*), tla.times(sets :_*))
+            // track the modification to point to the first variable and set
+            tracker.hold(vars.head, pair._1)
+            tracker.hold(sets.head, pair._2)
+            pair
+          } else {
+            (vars.head, sets.head)
```

Comment:
In this case, no.

Reply:
If "no" means "no, it's not impossible" then I guess we should handle this differently. Using pattern matching on the list instead of inspecting the index can help ensure all cases are covered safely in a concise way.

If "no" means, "no, it's not possible", then this convo can just be resolved ;)